### PR TITLE
feat: [CI-4621]: Support GIT API's via Token for Azure Repo for CI Manager

### PR DESCRIPTION
**Pull request by raghavharness**
## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/33884)

### CODE REVIEW

**Comment by raghavharness** [comment]
trigger messagecheck

**Comment by raghavharness** [comment]
trigger compile

**Comment by raghavharness** [comment]
trigger buildnum

**Comment by raghavharness** [comment]
trigger checkstyle

**Comment by raghavharness** [comment]
trigger pmd

File: 320-ci-execution/src/main/java/io/harness/ci/integrationstage/IntegrationStageUtils.java [hunk 1]
```diff
@@ -274,11 +279,21 @@ public String retrieveGenericGitConnectorURL(CodeBase ciCodebase, GitConnectionT
         throw new IllegalArgumentException("Repo name is not set in CI codebase spec");
       }
 
+      String projectName = ciCodebase.getProjectName().getValue();
       String repoName = ciCodebase.getRepoName().getValue();
-      if (url.endsWith(PATH_SEPARATOR)) {
-        gitUrl = url + repoName;
+
+      if (isNotEmpty(projectName) && url.contains(AZURE_REPO_BASE_URL)) {
```
Comment (harshjain12) [inline 320-ci-execution/src/main/java/io/harness/ci/integrationstage/IntegrationStageUtils.java:285]: Move it to new method

File: 320-ci-execution/src/main/java/io/harness/ci/integrationstage/IntegrationStageUtils.java [hunk 2]
```diff
@@ -274,11 +279,21 @@ public String retrieveGenericGitConnectorURL(CodeBase ciCodebase, GitConnectionT
...
+        gitUrl = StringUtils.join(StringUtils.stripEnd(url, PATH_SEPARATOR), PATH_SEPARATOR,
+            StringUtils.stripStart(projectName, PATH_SEPARATOR));
+        if (url.startsWith("git@ssh")) {
```
Comment (harshjain12) [inline 320-ci-execution/src/main/java/io/harness/ci/integrationstage/IntegrationStageUtils.java:288]: Use Group io.harness.git.GitClientHelper#GIT_URL_REGEX

File: 320-ci-execution/src/main/java/io/harness/ci/plan/creator/CIModuleInfoProvider.java [hunk 3]
```diff
@@ -375,4 +377,9 @@ private boolean isPrivateRepo(String urlString) {
       return true;
     }
   }
+
+  private boolean shouldReverseCommitList(String repoUrl) {
+    // Azure gives latest commit first
+    return !repoUrl.contains("dev.azure.com");
```
Comment (harshjain12) [inline 320-ci-execution/src/main/java/io/harness/ci/plan/creator/CIModuleInfoProvider.java:383]: remove hard coding

**Pull request merged**